Lib/test/test_asyncgen.py

```diff
@@ -377,6 +378,14 @@ async def async_gen_wrapper():
 
         self.compare_generators(sync_gen_wrapper(), async_gen_wrapper())
 
+    def test_async_gen_3_arg_deprecation_warning(self):
+        async def gen():
+            yield 123
+
+        with warnings.catch_warnings():
```

Comment:
This should test that `DeprecationWarning` is emitted rather than ignoring it. Same for all other places.

Reply:
Good point.

(One test for each of the 3 deprecated APIs should check that the deprecation warning is raised. Where the point of the test is to check something else, the deprecation warning can be suppressed).